classes/ShopMaintenance.php

```diff
@@ -126,66 +126,61 @@ public static function cleanAdminControllerMessages()
      *
      * @since 1.0.9
      */
-    public static function cleanImagesDir($new='') 
-	{				
-		if (isset($new) && $new=='new') {
-			$images_list = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS('SELECT id_image FROM '._DB_PREFIX_.'image ORDER BY id_image DESC');	
-			$imagesMax = $images_list[0]['id_image'];
-			Configuration::updateGlobalValue('SHOP_MAINTENANCE_IMAGES_RUN', $imagesMax.'|'.$imagesMax); 
-		}
-		else {
-			$imagesRun = Configuration::get('SHOP_MAINTENANCE_IMAGES_RUN');
-			$imgRun = explode("|", $imagesRun);
-			if (isset($imgRun[1])) $imagesMax = intval($imgRun[1]);
-				else $imagesMax = 0;	
-			if (isset($imgRun[0])) $imagesNow = intval($imgRun[0]);
-				else $imagesNow = 0;
-			if ($imagesMax > 0 && $imagesNow > 0) static::CheckAndFixImages($imagesNow, $imagesMax); 
-		}
-	}
-
- 	public static function CheckAndFixImages($imagesNow, $imagesMax)
+    public static function cleanImagesDir($new='')
     {
-		$starttime = time(); 
+	    if (isset($new) && $new=='new') {
+		    $images_list = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS('SELECT id_image FROM '._DB_PREFIX_.'image ORDER BY id_image DESC');
```

Comment:
To find the highest number of a database column, there's MAX(): https://dev.mysql.com/doc/refman/5.7/en/example-maximum-column.html. Shout be much less memory hungry than requesting a list with 100,000's of items.

Also, thirty bees prefers usage of DbQuery. Like
```php
$imagesMax = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS(
    (new DbQuery())
        ->select('MAX(`id_image`)')
        ->from('image')
);
```

Reply:
changes applied on the 92d8f9a updated commit.